contracts/MemeFactory.sol

```diff
@@ -381,15 +382,13 @@ abstract contract MemeFactory {
         // Get meme summon info
         MemeSummon storage memeSummon = memeSummons[memeNonce];
 
-        // Get the total native token committed to this meme
-        uint256 totalNativeTokenCommitted = memeSummon.nativeTokenContributed;
-
         // Check that the meme has been summoned
         require(memeSummon.summonTime > 0, "Meme not yet summoned");
         // Check if the token has been unleashed
         require(memeSummon.unleashTime == 0, "Meme already unleashed");
 
         // Update meme token map values
+        uint256 totalNativeTokenCommitted = memeSummon.nativeTokenContributed;
         totalNativeTokenCommitted += msg.value;
         memeSummon.nativeTokenContributed = totalNativeTokenCommitted;
```

Comment:
is this gas optimisation?

Reply:
yeah originally, but there are lots of places with state vars at play without optimizations, so might need to change to more readable or have an optimization pass everywhere later